community/kernel/src/main/java/org/neo4j/kernel/impl/coreapi/PropertyContainerLocker.java

```diff
@@ -69,6 +69,34 @@ void release( Statement statement, ResourceType type, long resourceId )
         }
     }
 
+    /**
+     * The Cypher runtime keeps statements open for longer, so this method does not close the statement after itself
+     */
+    public Lock exclusiveLock( Statement statement, PropertyContainer container )
+    {
+        if ( container instanceof Node )
```

Comment:
Not really - there are is a exclusive/share version of the methods that take a `Supplier<Statement>`, but only a single exclusive version that does not close the statement after itself.

I agree that it's not pretty - this is just a bug fix in the 2.3 maintenance branch. When merging into the 3.0 branch, we should just do a null-merge, because this has already been fixed "for reals" there